src/cfg/cfg-traversal.h

```diff
@@ -304,16 +341,20 @@ struct CFGWalker : public ControlFlowWalker<SubType, VisitorType> {
         break;
       }
       case Expression::Id::TryId: {
-        // FIXME Update the implementation to match the new spec
-        WASM_UNREACHABLE("unimp");
-        /*
         self->pushTask(SubType::doEndTry, currp);
-        self->pushTask(SubType::scan, &curr->cast<Try>()->catchBody);
-        self->pushTask(SubType::doStartCatch, currp);
+        auto& catchBodies = curr->cast<Try>()->catchBodies;
+        using namespace std::placeholders;
+        for (Index i = 0; i < catchBodies.size(); i++) {
+          auto doEndCatchI = std::bind(doEndCatch, _1, _2, i);
```

Comment:
Lambdas would be more readable than `std::bind` IMO.

Reply:
I agree, but lambdas capturing variables cannot be passed as a function argument. (Here we need to capture `i`) Do you know any way around it?

Reply:
Would capture-by-copy using `[i](...) { doEndCatch(..., i); }` work?

Reply:
That works, thanks. I was a little confused while experimenting with different settings; what didn't allow capturing lambdas to be passed as a function argument is the current `TaskFunc`, which is a function pointer. If we change `TaskFunc` to `std::function`, I mean, from
```cpp
typedef void (*TaskFunc)(SubType*, Expression**);
```
to
```cpp
using TaskFunc = std::function<void(SubType*, Expression**)>; 
```
This seems to be able to take capturing lambdas without problem.